Fixed Pagination

hemant10yadav opened the pull request:
## Technical Summary

This PR refactors the pagination component by replacing anchor tags with buttons. JavaScript is utilized to dynamically update and retrieve the most recent URL, addressing an issue where the `querystring` only retains parameters that were originally present in the URL.

Furthermore, the tab functionality has been adjusted to prevent the page parameter from being carried over when the user switches tabs manually.

[QA-7701](https://dimagi.atlassian.net/browse/QA-7701)

## Safety Assurance

### Safety story
Tested Locally


### Labels & Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change


[QA-7701]: https://dimagi.atlassian.net/browse/QA-7701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sravfeyn commented:
Thanks for the fix @hemant10yadav . Have you tested the pagination and sorting on all tables with this change applied?

hemant10yadav commented:
> Thanks for the fix @hemant10yadav . Have you tested the pagination and sorting on all tables with this change applied?

Yeah, I have tested it on the worker list page and the opportunity list page. It is working fine in both cases — where HTMX is used and where it is not used.

I also added a fix in this commit([64ba365](https://github.com/dimagi/commcare-connect/pull/595/commits/64ba36541024a15bfe706ee938bad7d460fe6c62)) and refactored the repetitive code.

sravfeyn reviewed (approved):
Looks good, but please merge only after testing on all tables (along with sorting)

sravfeyn reviewed (approved):
Looks good, but please merge only after testing on all tables (along with sorting)
